### ISSUE TITLE
Incorrect SQL constraint primary index name

### ISSUE THREAD

**Issue by Gerold103** [tarantool/tarantool#3788]
In https://github.com/tarantool/tarantool/issues/3503 an example is given:
```Lua
box.sql.execute("CREATE TABLE te10 (s1 INT, CONSTRAINT c PRIMARY KEY (s1), CONSTRAINT c CHECK (s1 > 0));")
```

It not only do not throw an error on constraint name duplicate, but also changes primary constraint name:
```Yaml
---
  - [512, 0, 'pk_C_1', 'tree', {'unique': true, 'sql': ''}, [{'sort_order': 'asc',
        'type': 'integer', 'field': 0, 'nullable_action': 'abort', 'is_nullable': false}]]
...
```

But it is expected that the primary index has name 'C' as it is specified explicitly.

**Comment by Korablev77** [comment]
>  If a user explicitly specifies index name via CONSTRAINT <name>, then he will expect to have this name both available via box.space.index.<name>

Why? As I already mentioned, index is only an implementation of unique constraint, so generally speaking creation of unique constraint may not lead to creation of index (I guess that all DBs follow this way, but it is not stated in standard). For instance, in MySQL (as far as I remember) creation of FK constraint also results in implicit creation of index, meanwhile in PostgreSQL it doesn't.
Still, this is very specific nitpicking and I guess we really should simply maintain only one global namespace of constraints.

**Comment by Gerold103** [comment]
You probably misunderstood my point. I mean, `CONSTRAINT <name> UNIQUE ...` should create an index with exactly name `<name>`, as well as `CONSTRAINT <name> PRIMARY KEY`. Foreign key should not create an index, and it already works ok - when you do `ALTER TABLE ... ADD CONSTRAINT <name>`, it is inserted into `_fk_constraint` with exactly the specified name. I do not want to lost `<name>` for a constraint, and want to make it visible via Lua from appropriate system spaces, `box.space` attributes depending on constraint type.

**Comment by pgulutzan** [comment]
I too probably misunderstood what Gerold103 said earlier:
"I vote for monolithic namespace of all constraints of one table: indexes, checks, foreign keys."
The constraint namespace should be for the whole schema, not merely one table.
Indexes are not constraints so I am only commenting about checks, foreign keys, etc.

**Comment by romanhabibov** [comment]
Closed via 1f783f268ff8b9076533388bf25e97da4bd592ed
